src/_pytest/raises.py

```diff
@@ -465,9 +465,9 @@ def _parse_exc(
                 )
         # unclear if the Type/ValueError distinction is even helpful here
         msg = f"expected exception must be {expected}, not "
-        if isinstance(exc, type):
+        if isinstance(exc, type):  # type: ignore[unreachable]
```

Comment:
Why not change the type in the method signature and rely on these to perform type narrowing?